mjpc/estimators/buffer.cc

```diff
@@ -49,66 +49,66 @@ void Buffer::Initialize(int dim_sensor, int num_sensor, int dim_ctrl,
   // sensor
 // reset
 void Buffer::Reset() {
-  sensor_.Reset();
-  sensor_.length_ = 0;
+  sensor.Reset();
+  sensor.SetLength(0);
 
   // sensor mask
-  sensor_mask_.Reset();
-
-  // TODO(etom): use a public interface:
-  sensor_mask_.length_ = 0;
+  sensor_mask.Reset();
+  sensor_mask.SetLength(0);
 
   // mask
   // ctrl
-  std::fill(mask_.begin(), mask_.end(), 1);  // set to true
+  std::fill(mask.begin(), mask.end(), 1);  // set to true
+  ctrl.Reset();
+  ctrl.SetLength(0);
 
-  ctrl_.Reset();
   // time
-  ctrl_.length_ = 0;
-
-  time_.Reset();
-  time_.length_ = 0;
+  time.Reset();
+  time.SetLength(0);
 }
 
 // update
 void Buffer::Update(const double* sensor, const int* mask, const double* ctrl,
                     double time) {
-  if (time_.length_ <= max_length_) {  // fill buffer
+  if (this->time.Length() <= max_length_) {  // fill buffer
```

Comment:
why do you need "this->" here?

Reply:
The class member has the same name as the local variable. https://www.geeksforgeeks.org/this-pointer-in-c/